In [1]:
import pandas as pd 
import numpy as np
from scipy import stats
from scipy.stats import norm

In [2]:
cutlets = pd.read_csv(r"C:\Cutlets.csv")

In [3]:
cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
cutlets.shape

(35, 2)

In [5]:
unitA  = pd.Series(cutlets.iloc[:,0])

In [6]:
unitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [7]:
unitB = pd.Series(cutlets.iloc[:,1])

In [8]:
unitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [9]:
p_value = stats.ttest_ind(unitA,unitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [10]:
p_value[1]

0.4722394724599501

In [11]:
p_value[0]

0.7228688704678063

# q2

In [12]:
lablat = pd.read_csv("C:\LabTAT.csv")

In [13]:
lablat

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [14]:
p_value = stats.f_oneway(lablat.iloc[:,0],lablat.iloc[:,1],lablat.iloc[:,2],lablat.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [15]:
p_value[1]

2.1156708949992414e-57

# QUESTION 3


In [16]:
from scipy import stats
from scipy.stats import norm,chi2_contingency
import numpy as np



In [17]:
buyerratio = pd.read_csv("C:\BuyerRatio.csv")

In [18]:
buyerratio.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [19]:
obs = np.array([[50,142,131,70],[435,1523,1356,750]])
obs

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [20]:
chi2_contingency(obs)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [21]:
#compare p-value with aplha = 0.05

## QUESTION 4

In [33]:
data = pd.read_csv("C:\Costomer+OrderForm.csv")

In [34]:
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [35]:
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [28]:
Phillippines = co.Phillippines.value_counts()
Indonesia = co.Indonesia.value_counts()
Malta = co.Malta.value_counts()
India = co.India.value_counts()

In [29]:
India

Error Free    280
Defective      20
Name: India, dtype: int64

In [36]:
data1 = pd.DataFrame(index=['Error Free','Defective'],
                   data = {'Phillippines':[271,29],
                        'Indonesia':[267,33],
                        'Malta': [269,31],
                        'India': [280,20]})
data1

Phillippines  Indonesia  Malta  India
Error Free           271        267    269    280
Defective             29         33     31     20

In [37]:
from scipy.stats import  chi2_contingency
p_value = stats.chi2_contingency(data1)

In [38]:
p_value

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [39]:
p_value[1]

0.2771020991233135

### P_VALUE(0.277)>APLHA(0.05), SO WE CANT REJECT THE NULL HYPOTHESIS

In [40]:
#PERCENTAGE OF DEFECTIVE ACROSS ALL THE CENTERS IS SAME